linera-base/src/data_types.rs

```diff
@@ -23,11 +23,38 @@ use crate::doc_scalar;
 pub struct Amount(u64);
 
 /// The balance of a chain.
-#[derive(
-    Eq, PartialEq, Ord, PartialOrd, Copy, Clone, Hash, Default, Debug, Serialize, Deserialize,
-)]
+#[derive(Eq, PartialEq, Ord, PartialOrd, Copy, Clone, Hash, Default, Debug)]
 pub struct Balance(u128);
 
+#[derive(Serialize, Deserialize)]
+struct BalanceU64 {
+    upper: u64,
+    lower: u64,
+}
+
```

Comment:
Nice! Couple of micro-optimizations:
* It does matter for JSON but just in case, make sure Serde sees the name "Balance" in the human-readable case by using `#[serde(rename = "Balance")]` before `struct BalanceU64 { .. }`
* For the binary case, it would be nice to also delegate (de)serialization to an aliased struct: `#[derive(Serialize, Deserialize)] #[serde(rename = "Balance")] struct BalanceU128(u128)` This would allow us to keep `formats.yaml` as it is.